src/hapi_schema/db_idps.py

```diff
@@ -0,0 +1,136 @@
+"""IDPs subcategory table and view."""
+
+from datetime import datetime
+
+from sqlalchemy import (
+    DateTime,
+    ForeignKey,
+    Integer,
+    select,
+)
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy.sql.expression import literal
+
+from hapi_schema.db_admin1 import DBAdmin1
+from hapi_schema.db_admin2 import DBAdmin2
+from hapi_schema.db_location import DBLocation
+from hapi_schema.db_resource import DBResource
+from hapi_schema.utils.base import Base
+from hapi_schema.utils.constraints import (
+    population_constraint,
+    reference_period_constraint,
+)
+from hapi_schema.utils.enums import DTMAssessmentType, build_enum_using_values
+from hapi_schema.utils.view_params import ViewParams
+
+
+class DBIDPs(Base):
+    __tablename__ = "idps"
+    __table_args__ = (
+        population_constraint(),
+        reference_period_constraint(),
+    )
+
+    resource_hdx_id = mapped_column(
+        ForeignKey("resource.hdx_id", onupdate="CASCADE", ondelete="CASCADE"),
+        nullable=False,
+    )
+
+    admin2_ref = mapped_column(
+        ForeignKey("admin2.id", onupdate="CASCADE"), primary_key=True
+    )
+
+    assessment_type: Mapped[DTMAssessmentType] = mapped_column(
+        build_enum_using_values(DTMAssessmentType), primary_key=True
+    )
+
+    reporting_round: Mapped[int] = mapped_column(Integer, nullable=True)
```

Comment:
There should be a >=1 constraint

Reply:
I think non_negative_constraint is probably better, in case some regions report 0 IDPs.  I'll add that.

Reply:
Oh, sorry, I misread.  Yes, I'll add that constraint.